### PR TITLE
Edit co insured presenter

### PR DESCRIPTION
**Pull request by hugokallstrom**

### CODE REVIEW

File: app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/data/GetCoInsuredUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import octopus.CoInsuredQuery
+
+internal interface GetCoInsuredUseCase {
+  suspend fun invoke(contractId: String): Either<CoInsuredError, CoInsuredResult>
+}
+
+internal class GetCoInsuredUseCaseImpl(
+  private val apolloClient: ApolloClient,
+) : GetCoInsuredUseCase {
+  override suspend fun invoke(contractId: String): Either<CoInsuredError, CoInsuredResult> = either {
```
Comment (hugokallstrom) [inline app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/data/GetCoInsuredUseCase.kt:22]: This is still WIP

**Review by StylianosGakis — APPROVED** [review]

File: app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/di/EditCoInsuredModule.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    GetCoInsuredUseCaseImpl(
+      get<ApolloClient>(octopusClient),
+    )
+  }
+
+  single<GetCoInsuredUseCaseProvider> {
+    GetCoInsuredUseCaseProvider(
+      demoManager = get<DemoManager>(),
+      prodImpl = get<GetCoInsuredUseCaseImpl>(),
+      demoImpl = get<GetCoInsuredUseCaseImpl>(),
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/di/EditCoInsuredModule.kt:22]: Yeah ill remove the providers for now

File: app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/EditCoInsuredPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+) : MoleculePresenter<EditCoInsuredEvent, EditCoInsuredState> {
+  @Composable
+  override fun MoleculePresenterScope<EditCoInsuredEvent>.present(lastState: EditCoInsuredState): EditCoInsuredState {
+    var errorMessage by remember { mutableStateOf(lastState.errorMessage) }
+    var isLoading by remember { mutableStateOf(lastState.isLoading) }
+    var coInsured by remember { mutableStateOf(lastState.coInsured) }
+    LaunchedEffect(Unit) {
+      Snapshot.withMutableSnapshot {
+        isLoading = true
+      }
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/EditCoInsuredPresenter.kt:31]: I see, thanks!

File: app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/EditCoInsuredPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import com.hedvig.android.feature.editcoinsured.data.GetCoInsuredUseCase
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.persistentListOf
+
+internal class EditCoInsuredPresenter(
+  private val contractId: String,
+  private val getCoInsuredUseCaseProvider: Provider<GetCoInsuredUseCase>,
+) : MoleculePresenter<EditCoInsuredEvent, EditCoInsuredState> {
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/EditCoInsuredPresenter.kt:22]: Events will come in future PRs on top :) Didnt want to make this too big. Should add TODOs.

**Pull request merged**